src/library_manager/lib_manager.c

```diff
@@ -487,15 +487,28 @@ static void lib_manager_update_sof_ctx(void *base_addr, uint32_t lib_id)
 }
 
 #if CONFIG_INTEL_MODULES
-int lib_manager_register_module(struct sof_man_fw_desc *desc, int module_id)
+int lib_manager_register_module(const uint32_t component_id)
 {
-	uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(module_id);
-	/* allocate new  comp_driver_info */
+	const struct lib_manager_mod_ctx *const ctx = lib_manager_get_mod_ctx(component_id);
+	const uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(component_id);
+	const struct sof_man_fw_desc *desc;
 	struct comp_driver_info *new_drv_info;
 	struct comp_driver *drv = NULL;
 	struct sof_man_module *mod;
 	int ret;
 
+	/* Get library manifest based on component_id */
+	if (!ctx || !ctx->base_addr)
+		return -ENOENT;
+
+	desc = (const struct sof_man_fw_desc *)((const char *)ctx->base_addr +
+							      SOF_MAN_ELF_TEXT_OFFSET);
```

Comment:
can we use `lib_manager_get_library_module_desc()`?

Reply:
I will use use `ctx` to get buildinfo in future PR.